Address Magali's comments

lhy11009 opened the pull request:
I also removed the 'todo' tags

elodie-kendall committed:
add option for different plasticity for different phases

rfildes committed:
adding static temperature field

elodie-kendall committed:
add test

elodie-kendall committed:
address comments

anne-glerum committed:
Add viscous dissipation postprocessors

anne-glerum committed:
Add postprocessor to existing test

jdannberg committed:
add christensen yuen cookbook

jdannberg committed:
address comments and update tests

jdannberg committed:
address more comments

tjhei committed:
fix benchmarks/newton/spiegelman

fixes
An error occurred in line <174> of file
<../source/boundary_temperature/interface.cc> in function
void
aspect::BoundaryTemperature::Manager<dim>::parse_parameters(dealii::ParameterHandler&)
[with int dim = 2]
The violated condition was:
    model_names.size() == 0
Additional information:
You have indicated that you wish to apply a boundary temperature
model, but the <Fixed temperature boundary indicators> parameter is
empty. Please use this parameter to specify the boundaries on which the
model(s) should be applied.

cedrict committed:
avoiding division by zero in dyn topo pp

cedrict committed:
add 6 names to coverpage bottom

gassmoeller committed:
Unify particle update for deal.II 9.2 and 9.3

tjhei committed:
Merge pull request #4240 from tjhei/fix_spiegelman_prm

fix benchmarks/newton/spiegelman

gassmoeller committed:
Only use optimizations when supported.

jaustermann committed:
fix a typo

jdannberg committed:
Merge pull request #4245 from jaustermann/fix_typo

fix a typo

zjiaqi2018 committed:
GMG Newton solver

tjhei committed:
Merge pull request #4243 from gassmoeller/unify_handling_of_particle_updates

Unify particle update for deal.II 9.2 and 9.3

MFraters committed:
Merge pull request #4241 from cedrict/dyn_topo_no_div_by_zero

avoiding division by zero in dyn topo pp

MFraters committed:
Merge pull request #4242 from cedrict/add_names_coverpage

add 6 names to coverpage bottom

bobmyhill committed:
Merge pull request #4228 from jdannberg/christensen_yuen

add christensen yuen cookbook

anne-glerum committed:
Merge postprocessors and update test

sibiaoliu committed:
Allow to use adiabatic pressure for creep viscosity calculation in visco-plastic rheology

anne-glerum committed:
Update test

sibiaoliu committed:
add a test for adiabatic P in vp model

address comments

tjhei committed:
fix crameri

tjhei committed:
fix inclusion

naliboff committed:
Merge pull request #4161 from elodie-kendall/aspect-2.3pre_plasticityphases_no1900file

add option for different plasticity for different phases

tjhei committed:
GMG: implement mask for velocity boundary

tjhei committed:
Merge pull request #4250 from tjhei/fix_spiegelman_prm

Fix benchmarks/

tjhei committed:
let the test use GMG

sibiaoliu committed:
address comments; update test results; indentation

tjhei committed:
update test results

mibillen commented on doc/manual/cookbooks/phase_diagram/doc/phase_diagram.tex:
```diff
@@ -22,7 +22,9 @@ \subsubsection{Phase diagram}
 \par The material model is then set up to mimic mantle phase transitions at 410, 520, and 660 km.% setup of mantle transitions
 Details of phase transitions are taken from \cite{billen2018decoupling}.
-A trick of assigning the values of density to the field of thermal capacity is used here [MIB: How is this done? can you be more explicit]. % trick
+A trick of assigning the values of density to the field of thermal capacity is used here. % [MIB: How is this done? can you be more explicit, addressed]. % trick
```
Can you change this to:  ""### 
```suggestion
A trick is needed to make this work: I assign the values of the density to the field of thermal capacity in the input file. % trick
```

mibillen commented on doc/manual/cookbooks/phase_diagram/doc/phase_diagram.tex:
```diff
@@ -22,7 +22,9 @@ \subsubsection{Phase diagram}
-A trick of assigning the values of density to the field of thermal capacity is used here [MIB: How is this done? can you be more explicit]. % trick
+A trick of assigning the values of density to the field of thermal capacity is used here. % [MIB: How is this done? can you be more explicit, addressed]. % trick
+To do this, I used the phase inputs implemented in the ViscoPlastic file.
+There, one can either assign values to individual phases with the separator '|', or assign constant values to all phases by a single number.
```
I don't think this line (27) is actually needed. I was confused about where the trick in line 25 was happening. 
```suggestion
```

mibillen reviewed (commented):
Just one line to fix.

tjhei committed:
Merge pull request #4249 from gassmoeller/revert_some_particle_changes

Revert some particle performance regressions

gassmoeller committed:
Merge pull request #4152 from rfildes/static_temperature

Static temperature

gassmoeller committed:
Merge pull request #4251 from tjhei/gmg_velocity_boundary_component

GMG: implement mask for velocity boundary

bangerth committed:
Don't use std::make_unique just yet.

tjhei committed:
Merge pull request #4261 from bangerth/make_unique

Don't use std::make_unique just yet.

naliboff committed:
check DG limiters only used with cartesian mapping

naliboff committed:
Merge pull request #4247 from sibiaoliu/add_enable_adiabaticP_in_creepflow

Allow to use adiabatic P for creep viscosity calculation in visco-plastic rheology

zjiaqi2018 committed:
address comments

zjiaqi2018 committed:
update test results

naliboff committed:
restructure assert and cleanup

Co-authored-by: Wolfgang Bangerth <bangerth@colostate.edu>

gassmoeller committed:
Merge pull request #4214 from anne-glerum/viscous_dissipation_postprocessor

Viscous dissipation postprocessors

gassmoeller committed:
Minor rewording

tjhei committed:
Merge pull request #4217 from zjiaqi2018/gmg-newton

GMG Newton solver

tjhei committed:
Merge pull request #4262 from naliboff/disable_dg_limiter_spherical_geometry

check DG limiters only used with cartesian mapping

bangerth committed:
Merge pull request #4265 from gassmoeller/minor_tweaks_viscous_dissipation_statistics

Minor refactoring of viscous dissipation statistics

bangerth committed:
Remove some unnecessary include files.

bangerth committed:
Rename an argument.

The previous name 'use_strain_rate' did not quite describe what the argument
represented. Better to call it 'compute_strain_rate'. While there, document
what that argument actually means.

bangerth committed:
Fix a possible bug.

It's not wrong as is right now because the velocity always comes first among
all variables, but the code right now makes an assumption that is unnecessary.
The author of this code got it right in the next line already :-)

tjhei committed:
Merge pull request #4268 from bangerth/bug

Fix a possible bug.

tjhei committed:
Merge pull request #4266 from bangerth/includes

Remove some unnecessary include files.

erinheilman committed:
Update strain healing tests

erinheilman committed:
fix indentation

gassmoeller committed:
Disable failing particle integration schemes in periodic shells

bangerth committed:
Merge pull request #4272 from gassmoeller/disable_broken_integrators_periodic_sphere

Disable failing particle integration schemes in periodic shells

tjhei committed:
remove PETSc cmake and misc

remove PETSc from cmake, manual, CI

tjhei committed:
update zenodo DOI

tjhei committed:
Merge pull request #4267 from bangerth/rename

Rename an argument.

MFraters committed:
Merge pull request #4278 from tjhei/readme-zenodo

update zenodo DOI in readme

tjhei committed:
add authors to zenodo template

tjhei committed:
add/update bib entries for release

bangerth committed:
Merge pull request #4276 from tjhei/rm_cmake_petsc

remove PETSc cmake and misc

bangerth committed:
Remove remaining uses of PETSc from the code base.

bangerth committed:
Merge pull request #4280 from tjhei/update_bib_entries_for_release

add/update bib entries for release

bangerth committed:
Merge pull request #4279 from tjhei/zenodo-authors

add maintainers to zenodo template

bangerth committed:
Use a proper reference.

bangerth committed:
Use \[...\] instead of $$...$$.

tjhei committed:
Merge pull request #4284 from bangerth/ref

Use a proper reference.

tjhei committed:
Merge pull request #4283 from bangerth/petsc

Remove remaining uses of PETSc from the code base.

bangerth committed:
Minor grammar fixes.

bangerth committed:
Merge pull request #4271 from erinheilman/update_strain_healing_field_tests

Update strain healing field tests

tjhei committed:
Merge pull request #4285 from bangerth/latex

Use \[...\] instead of $$...$$.

tjhei committed:
Merge pull request #4287 from bangerth/formatting

Minor grammar fixes.

lhy11009 committed:
Add cookbook for phase diagram
